npm/oxlint/package.json

```diff
@@ -25,6 +25,9 @@
   "engines": {
     "node": "^20.19.0 || >=22.12.0"
   },
+  "optionalDependencies": {
+    "typescript": ">=4.8.4 <6.0.0"
+  },
```

Comment:
As discussed, unfortunately I think we need to promote `typescript` to a full `dependency`. We can also then tighten the version range to whatever is latest version.